src/system/storage/storage.cpp

```diff
@@ -59,25 +86,80 @@ bool Storage::saveConfiguration(int relayID, uint8_t hour, uint8_t minute, uint8
 
 bool Storage::getCredentials(String templateID, String templateName, String authToken)
 {
-  EEPROM.get(offsetof(EEPROM_CredentialStorage_t, templateid),templateID);
-  EEPROM.get(offsetof(EEPROM_CredentialStorage_t, templateName),templateName);
-  EEPROM.get(offsetof(EEPROM_CredentialStorage_t, authToken),authToken);
-
-  logger << LOG_INFO << "templateID " << templateID << EndLine;
-  logger << LOG_INFO << "templateName " << templateName << EndLine;
-  logger << LOG_INFO << "authToken " << authToken << EndLine;
+  EEPROM_CredentialStorage_t d;
+  // Check if info was saved previously, if not return empty strings
+  if (getPrevSavedInfo())
+  {
+    // Credentials are on the bottom of the reserved memory, use address 0
+    EEPROM.get(0, d);
+    templateID = d._templateid;
+    templateName = d._templateName;
+    authToken = d._authToken;
 
+    logger << LOG_INFO << "templateID " << templateID << EndLine;
+    logger << LOG_INFO << "templateName " << templateName << EndLine;
+    logger << LOG_INFO << "authToken " << authToken << EndLine;
+  }
+  else
+  {
+    templateID = templateName = authToken = "";
+  }
   return true;
 }
 
 bool Storage::getConfiguration(int relayID, uint8_t hour, uint8_t minute, uint8_t second, uint8_t duration, uint8_t days)
 {
   EEPROM.get(CONF_MEM_START + (sizeof(eeprom_map_conf_time_t) * relayID), relayConfigTime);
-  hour      = relayConfigTime.hour;
-  minute    = relayConfigTime.min;
-  second    = relayConfigTime.sec;
-  duration  = relayConfigTime.duration;
-  days      = relayConfigTime.days;
+  hour = relayConfigTime.hour;
+  minute = relayConfigTime.min;
+  second = relayConfigTime.sec;
+  duration = relayConfigTime.duration;
+  days = relayConfigTime.days;
 
   return true;
 }
+
+bool Storage::getPrevSavedInfo()
+{
+  uint8_t isSaved = 0;
+  EEPROM.get(SAVED_FLAG_START, isSaved);
+  logger << LOG_INFO << "Checking saved flag = " << INT_HEX << isSaved << EndLine;
+  if (SAVED_DATA_FLAG == isSaved)
+  {
+    return true;
+  }
+  return false;

```

Comment:
```suggestion
  return !(SAVED_DATA_FLAG == isSaved);
```

Reply:
shouldn´t be 
``return (SAVED_DATA_FLAG == isSaved)``
without the !

Reply:
Right